### PR TITLE
Add extras attribute to app_commands.Command

### PR DESCRIPTION
**Pull request by nerdguyahmad**
## Summary

This pull request adds an `extras` attribute to `app_commands.Command`

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/app_commands/commands.py [hunk 1]
```diff
@@ -492,6 +492,8 @@ class Command(Generic[GroupT, P, T]):
         Due to a Discord limitation, this does not work on subcommands.
     parent: Optional[:class:`Group`]
         The parent application command. ``None`` if there isn't one.
+    extras: :class:`dict`
+        A dictionary of user provided extras to attach to this command.
```
Comment (PaulMarisOUMary) [inline discord/app_commands/commands.py:496]:
```suggestion
        A dictionary that can be used to store extraneous data.
        The library will not touch any values or keys within this dictionary.
```

File: discord/app_commands/commands.py [hunk 2]
```diff
@@ -1527,6 +1532,8 @@ def command(
             of the callback shortened to 100 characters.
         nsfw: :class:`bool`
             Whether the command is NSFW and should only work in NSFW channels. Defaults to ``False``.
+        extras: :class:`dict`
+            A dictionary of user provided extras to attach to this command.
```
Comment (PaulMarisOUMary) [inline discord/app_commands/commands.py:1536]:
```suggestion
            A dictionary that can be used to store extraneous data.
            The library will not touch any values or keys within this dictionary.
```

File: discord/app_commands/commands.py [hunk 3]
```diff
@@ -1575,6 +1584,8 @@ def command(
         Whether the command is NSFW and should only work in NSFW channels. Defaults to ``False``.
 
         Due to a Discord limitation, this does not work on subcommands.
+    extras: :class:`dict`
+        A dictionary of user provided extras to attach to this command.
```
Comment (PaulMarisOUMary) [inline discord/app_commands/commands.py:1588]:
```suggestion
        A dictionary that can be used to store extraneous data.
        The library will not touch any values or keys within this dictionary.
```

File: discord/app_commands/tree.py [hunk 4]
```diff
@@ -860,6 +861,8 @@ def command(
             The list of guilds to add the command to. This cannot be mixed
             with the ``guild`` parameter. If no guilds are given at all
             then it becomes a global command instead.
+        extras: :class:`dict`
+            A dictionary of user provided extras to attach to this command.
```
Comment (PaulMarisOUMary) [inline discord/app_commands/tree.py:865]:
```suggestion
            A dictionary that can be used to store extraneous data.
            The library will not touch any values or keys within this dictionary.
```

**Review by PaulMarisOUMary — COMMENTED** [review]
Following [discord/interaction.py](https://github.com/Rapptz/discord.py/blob/573bb32a1e31ade15677eb2a945f81729646b546/discord/interactions.py#L115-L118)
https://github.com/Rapptz/discord.py/blob/573bb32a1e31ade15677eb2a945f81729646b546/discord/interactions.py#L115-L118
Moreover, it should be clearer from a developer's point of view reading the docs.

**Review by Rapptz — CHANGES_REQUESTED** [review]
These need to be copied in the various copy methods.

**Review by Rapptz — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**